lib/custom_lib/CMakeLists.txt

```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2021, Legrand North America, LLC.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_include_directories_ifdef(CONFIG_CUSTOM_LIB .)
```

Comment:
I've discussed this extensively with @tejlmand and we have come up with the following solution, which is cleaner for the current use-case provided

First, let us move the header to `include/custom_lib/custom_lib.h`. This is for consistency with the upstream zephyr tree, like we did for many others. Public headers that are to be used by the app belong in `include/`.

We first evaluated the possibility of doing this in pure CMake:
```
add_library(custom_lib STATIC)
target_sources(custom_lib PRIVATE custom_lib.c)
target_include_directories(custom_lib PUBLIC ${ZEPHYR_CURRENT_MODULE_DIR}/include/custom_lib)
target_link_libraries(custom_lib PRIVATE zephyr_interface)
target_link_libraries(app PRIVATE custom_lib)
```
which has the advantage of not polluting the zephyr build with the `custom_lib.h` include when building the in-tree zephyr modules.

But in the end we think it's simpler for this case to just use zephyr primitives until we decide exactly how to clean this up:

```
zephyr_library()
zephyr_library_sources(custom_lib.c)
zephyr_include_directories(${ZEPHYR_CURRENT_MODULE_DIR}/include/custom_lib)

```

Reply:
Accepted, and rolled in.